Sources/Holocron/StorageProvider+Default.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  StorageProvider+Default.swift
+//  Holocron-iOS
+//
+//  Created by Pranjal Satija on 4/15/19.
+//  Copyright © 2019 CocoaPods. All rights reserved.
+//
+
+// MARK: Default Coding
+struct Box<T> {
+    let element: T
+}
+
+extension Box: Decodable where T: Decodable { }
+extension Box: Encodable where T: Encodable { }
+
+extension StorageProvider {
+    /// Attempts to decode an object that was encoded using `defaultEncoded`.
+    /// - parameter data: The data to decode.
+    /// - returns: A decoded object.
+    func defaultDecoded<T: Decodable>(_ data: Data) throws -> T {
+        return try JSONDecoder().decode(Box<T>.self, from: data).element
```

Comment:
I don't know if we want to concern ourselves with migration. But using the `Box` has a few additional things we need the consider. It's only really needed for Int / String / Bool (essentially `Codable` things that will result in a `SingleValueContainer`. So if the previously used method of writing is not nesting `KeyedDecodingContainer` `Codable` things in a`Box` (your average `struct`/`class`), we'll have a backwards compatibility issue.

Reply:
I actually went with the `Box` because it looks like the old version of the library [did it](https://github.com/BottleRocketStudios/iOS-Holocron/blob/3c02613c7b6f608f177677e1babd0de6dce1ad90/Sources/Holocron/StorageContainer.swift#L96) and I didn't want to break compatibility.

Reply:
Sorry, I'm talking about non-Holocron backwards compatibility. Again, not sure if we care, but figured I'd bring it up 🤷‍♂️